metacat-thrift/src/main/java/com/netflix/metacat/thrift/CatalogThriftHiveMetastore.java

```diff
@@ -1162,17 +1163,26 @@ public Partition get_partition_with_auth(
     public PartitionsByExprResult get_partitions_by_expr(final PartitionsByExprRequest req) throws TException {
         return requestWrapper("get_partitions_by_expr", new Object[]{req},
             () -> {
-                String filter = null;
-                if (req.getExpr() != null) {
-                    filter = Utilities.deserializeExpressionFromKryo(req.getExpr()).getExprString();
-                    if (filter == null) {
-                        throw new MetaException("Failed to deserialize expression - ExprNodeDesc not present");
+                try {
+                    String filter = null;
+                    if (req.getExpr() != null) {
+                        filter = Utilities.deserializeExpressionFromKryo(req.getExpr()).getExprString();
+                        if (filter == null) {
+                            throw new MetaException("Failed to deserialize expression - ExprNodeDesc not present");
+                        }
                     }
+                    //TODO: We need to handle the case for 'hasUnknownPartitions'
+                    return new PartitionsByExprResult(
+                        getPartitionsByFilter(req.getDbName(), req.getTblName(), filter, req.getMaxParts()),
+                        false);
+                } catch (Exception e) {
+                    //
+                    // Added this to specially handle filter expression failures.
+                    // Extending this to any exception because Hive client then ignores the exception and executes the
+                    // query without predicate pruning.
+                    //
```

Comment:
are you saying the hive client from the thrift side? not sure I get it, since I thought hive client isn't in the loop after we roll out.

Reply:
We see this issue when running hive queries. Hive client instead of talking to RDS or HCAT directly, it can talk to Metacat.